src/window.rs

```diff
@@ -23,53 +47,93 @@ pub fn display_image(title: &str, image: &RgbaImage, window_width: u32, window_h
     let sdl = sdl2::init().expect("couldn't create sdl2 context");
     let video_subsystem = sdl.video().expect("couldn't create video subsystem");
 
-    let mut window = video_subsystem
-        .window(title, window_width, window_height)
-        .position_centered()
-        .resizable()
-        .build()
-        .expect("couldn't create window");
-
-    window
-        .set_minimum_size(MIN_WINDOW_DIMENSION, MIN_WINDOW_DIMENSION)
-        .expect("invalid minimum size for window");
+    let mut windows: Vec<sdl2::video::Window> = Vec::with_capacity(images.len());
+    let mut window_visibility: Vec<bool> = Vec::with_capacity(images.len());
+    for _ in 0..images.len() {
+        let mut window = video_subsystem
+            .window(title, window_width, window_height)
+            .position_centered()
+            .resizable()
+            .build()
+            .expect("couldn't create window");
+
```

Comment:
Creating all of the images in the same location is a bit confusing. You could use `set_position` to stagger each window by some multiple of its index.

Reply:
I have them staggered, however there is an issue if there are too many windows it is possible that they are rendered off screen. So i'll have to add some code to prevent that. And by too many windows, I mean around 20 and more.